include/precedence.hpp

```diff
@@ -62,13 +64,19 @@ namespace NP {
 			return fromIndex;
 		}
 
+		bool should_signal_at_completion() const
+		{
+			return signal_at_completion;
+		}
+
 	private:
 		JobID from;
 		JobID to;
 		// toIndex and fromIndex are set during validation
 		Job_index toIndex;
 		Job_index fromIndex;
 		Interval<Time> sus_times;
+		bool signal_at_completion;
```

Comment:
I propose to define an `enum` called `Prec_reference` containing enumarators `start` and `finish` instead of using a `bool`